### DISCUSSION TITLE
Can I configure APM for Redis, Mongodb on Skywalking ?

### DISCUSSION THREAD

**Discussion by nthoa12** [apache/skywalking#8033] [Q&A] [answered]
I want to view APM of Bigdata stacks on Skywalking. I have successfully configured some stack with javaagent such as Kafka, Solr, Elasticsearch. Does Skywalking have support for Redis, Mongodb? Thanks.

**Comment by wu-sheng** [comment, marked as answer]
I am not 100% sure what you are asking. What is agent on Kafka or ElasticSearch? We never recommended agent on the server side. All plugins are client libraries related, beside some special plugin for DolphinScheduler and Seata server. Their community build specific plugins for their server sides.
Maybe you misread some blog or something?

**Reply by chicknsoup** [reply]
> 
> 
> I am not 100% sure what you are asking. What is agent on Kafka or ElasticSearch? We never recommended agent on the server side. All plugins are client libraries related, beside some special plugin for DolphinScheduler and Seata server. Their community build specific plugins for their server sides. Maybe you misread some blog or something?

Maybe he wants to have some kinds of agent to inject to Redis/Mongodb server like the java agent (which can be used within Kafka broker or Elasticsearch server).

**Reply by wu-sheng** [reply]
We never tested agent for ElasticSearch or Kafka. If it works, then good to know.
AFAIK, redis server is even not a Java server, so, how is that possible?

**Reply by chicknsoup** [reply]
Idk, a super hard method like "code injection" maybe :)

**Reply by wu-sheng** [reply]
This is out community scope. They are open source, you can do on your own.

**Reply by nthoa12** [reply]
Thank @chicknsoup and @wu-sheng
